core/src/test/scala/org/apache/spark/deploy/history/HistoryServerSuite.scala

```diff
@@ -100,6 +100,7 @@ class HistoryServerSuite extends SparkFunSuite with BeforeAndAfter with Matchers
     "minDate app list json" -> "applications?minDate=2015-02-10",
     "maxDate app list json" -> "applications?maxDate=2015-02-10",
     "maxDate2 app list json" -> "applications?maxDate=2015-02-03T16:42:40.000GMT",
+    "limit app list json" -> "applications?limit=3",
```

Comment:
limit=-1 will actually return the whole list, I just pass the int to `take` and that's how that function works, I though this was a good default value (which is why it's the default value in the js file)